File: src/main/java/dev/toastmc/toastclient/api/setting/types/BooleanSetting.java
package dev.toastmc.toastclient.api.setting.types;

/**
 * Interface representing a boolean value that can be toggled.
 * @author lukflug
 */
public interface BooleanSetting {
    /**
     * Toggle the boolean value.
     */
    public void toggle();

    /**
     * Get the boolean value.
     * @return the value
     */
    public boolean enabled();

}

File: src/main/kotlin/dev/toastmc/toastclient/api/setting/types/BooleanSetting.kt
package dev.toastmc.toastclient.api.setting.types

/**
 * Interface representing a boolean value that can be toggled.
 * @author lukflug
 */
interface BooleanSetting {

    /**
     * Toggle the boolean value.
     */
    fun toggle()

    /**
     * Get the boolean value.
     * @return the value
     */
    fun enabled(): Boolean
}